sdk/src/utils/request.ts

```diff
@@ -21,8 +21,9 @@ export const request = async <Output = any>(
 
 	return fetcher(url, options).then((response) => {
 		return extractData(response).catch((reason) => {
-			const errors = typeof reason === 'object' && 'errors' in reason ? reason.errors : reason;
-			return Promise.reject({ errors, response });
+			const errors = reason && typeof reason === 'object' && 'errors' in reason ? reason.errors : reason;
+			const data = reason && typeof reason === 'object' && 'data' in reason ? reason.data : null;
+			return Promise.reject({ data, errors, response });
```

Comment:
I think we should only return the `data` prop if it is actually available in `reason`

```suggestion
			const result: { response: unknown; errors: any; data?: any } = {
				errors: reason && typeof reason === 'object' && 'errors' in reason ? reason.errors : reason,
				response,
			};

			if (reason && typeof reason === 'object' && 'data' in reason) result.data = reason.data;

			return Promise.reject(result);
```

Reply:
That makes sense to me! I was thinking about accessing it like `err.data == null` which would be the same, but I forgot to think about property checks like `Object.hasOwn`